packages/ember-metal/lib/property_events.js

```diff
@@ -112,10 +113,12 @@ function propertyDidChange(obj, keyName, _meta) {
 
   markObjectAsDirty(meta, keyName);
 
-  if (isEnabled('ember-glimmer-detect-backtracking-rerender') ||
+  runInDebug(() => {
+    if (isEnabled('ember-glimmer-detect-backtracking-rerender') ||
       isEnabled('ember-glimmer-allow-backtracking-rerender')) {
-    assertNotRendered(obj, keyName, meta);
-  }
+      assertNotRendered(obj, keyName, meta)
```

Comment:
So I dug into this a bit, and it turns out that this isn't the right fix. Specifically, in the case where `ember-glimmer-allow-backtracking-rerender` is enabled we have to actually call `assertNotRendered` or we will not trigger a rerender when we hit this backtracking.  In other words, wrapping this in `runInDebug` does avoid an error (due to a bug in `assertNotRendered`) but it also breaks the ability for rerendering to happen.


I submitted https://github.com/emberjs/ember.js/pull/14876 to address.